### ISSUE TITLE
Should uWSGI ignore empty TMPDIR environment variable?

### ISSUE THREAD

**Issue by abdrabo** [unbit/uwsgi#1966]
I faced the following error while `TMPDIR` was set to empty string on my environment:
`uwsgi_postbuffer_do_in_disk()/uwsgi_tmpfile(): No such file or directory [core/reader.c line 535] during POST /my/endpoint`

I tracked it down to the following part:
https://github.com/unbit/uwsgi/blob/8c811227165d5223d3edd7df34a622a391d3c8bc/core/utils.c#L3596-L3599

Essentially uWSGI will try to open/unlink tmp files in the root directory. This is not how `mktemp` works for example.

**Comment by xrmx** [comment]
Are you sure it was an empty string? In that case the check at 3597 should suffice no?

**Comment by abdrabo** [comment]
No, it not enough. Check the following and notice how FOO (set and empty) is different from BAR (unset)
```
$ cat test.c 
#include <stdio.h>
#include <stdlib.h>

char* f(char* key) {
    char* val = getenv(key);
    if (!val) return "/tmp";
    return val;
}

int main() {
    printf("FOO: '%s'\n", f("FOO"));
    printf("BAR: '%s'\n", f("BAR"));
}
$ gcc test.c 
$ FOO= ./a.out 
FOO: ''
BAR: '/tmp'
```

**Comment by xrmx** [comment]
ah of course it's a pointer :)

**Comment by abdrabo** [comment]
So back to the original question, should uWSGI ignore the empty TMPDIR? I can create a pull request.
